gem/optimise.py

```diff
@@ -242,6 +242,15 @@ def delta_elimination(sum_indices, factors):
     """
     sum_indices = list(sum_indices)  # copy for modification
 
+    def substitute(expression, from_, to_):
+        if from_ not in expression.free_indices:
+            return expression
+        elif isinstance(expression, Delta):
+            mapper = MemoizerArg(filtered_replace_indices)
+            return mapper(expression, ((from_, to_),))
```

Comment:
Yes, because if we allow to wrap `Delta` nodes with `Indexed(ComponentTensor(..., ...), ...)` then `isinstance(f, Delta)` will fail in the next round of the fixed point iteration looking for delta cancellation opportunities. See the test case.